### PR TITLE
Update training YAMLs with new defaults

### PR DESCRIPTION
**Pull request by abhi-mosaic**
FSDP new defaults:
* mixed_precision: PURE
* activation_checkpointing_reentrant: false
* limit_all_gathers: true

MosaicGPT new defaults:
* attn_impl='triton'
* param_init_fn='kaiming_normal_'
* init_nonlinearity='relu'
* low_precision_layernorm=True

Training loop new defaults
* `eval_first: false` just to expose that this an option

### CODE REVIEW

File: examples/llm/mcloud/mcli-1b-eval.yaml
Comment (vchiley) [inline examples/llm/mcloud/mcli-1b-eval.yaml:1]: `mixed_precision: PURE` in fsdp config?

do `${max_seq_len}` and `${tokenizer_name}`?
(gotta set it as well)

File: examples/llm/icl_eval/yamls/mosaic_gpt_eval.yaml [hunk 1]
```diff
@@ -13,18 +13,15 @@ model:
   n_layers: 12
   mlp_ratio: 4
   max_seq_len: 2048
```
Comment (vchiley) [inline examples/llm/icl_eval/yamls/mosaic_gpt_eval.yaml:15]: do `${max_seq_len}` instead?
(gotta set it above tho)

also `${tokenizer_name}`.

File: examples/llm/yamls/mosaic_gpt/1b.yaml [hunk 2]
```diff
@@ -18,11 +18,8 @@ model:
   mlp_ratio: 4
   max_seq_len: ${max_seq_len}
   vocab_size: 50368
-  init_std: 0.02
-  attn_pdrop: 0.0
-  resid_pdrop: 0.0
-  emb_pdrop: 0.0
-  attn_impl: flash
+
```
Comment (vchiley) [inline examples/llm/yamls/mosaic_gpt/1b.yaml:21]:
```suggestion
```

**Review by vchiley — APPROVED** [review]
added a few nit cmts.

also are we skipping `low_precision_layernorm: true`?

File: examples/llm/icl_eval/yamls/mosaic_gpt_eval.yaml (hunk 1 above)
Comment (abhi-mosaic) [reply examples/llm/icl_eval/yamls/mosaic_gpt_eval.yaml:15]: Oh this is actually a bigger issue b/c right now these values reference the `v0.0.3` branch. At the end of all `v0.0.4` changes we will have to self-reference the tag. I will update the stuff you mentioned now but this is expected to be broken on the release branch right now.

**Comment by abhi-mosaic** [comment]
> also are we skipping low_precision_layernorm: true?
I put this in the default `MosaicGPTConfig` now, so it should be true by default.

**Pull request merged**